From db826a7a68cd1afbae824a7addb3ee4624635dd9 Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Wed, 17 Aug 2022 15:27:50 +0200
Subject: [PATCH 28/30] disable Apple libc++ Availability tests

if we are using MacPort's installed libc++
then Apple's availablilty tests for libcxx by OS version
are no longer applicable
---
 libcxx/include/__config | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libcxx/include/__config b/libcxx/include/__config
index 575147cead42..24cda6ddc31b 100644
--- a/libcxx/include/__config
+++ b/libcxx/include/__config
@@ -10,6 +10,10 @@
 #ifndef _LIBCPP_CONFIG
 #define _LIBCPP_CONFIG
 
+// for MacPorts, our installed libcxx does not require Apple system
+// Availability tests
+#define _LIBCPP_DISABLE_AVAILABILITY
+
 #if defined(_MSC_VER) && !defined(__clang__)
 #  if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
 #    define _LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER
-- 
2.37.2

